test/e2e/parallel/spacebindingrequest_test.go

```diff
@@ -0,0 +1,267 @@
+			// wait for the namespace to be provisioned since we will be creating the SpaceBindingRequest into it.
+			space, err := hostAwait.WaitForSpace(t, space.Name, UntilSpaceHasAnyProvisionedNamespaces())
+			require.NoError(t, err)
+			// create the spacebinding request
+			spaceBindingRequest := CreateSpaceBindingRequest(t, awaitilities, memberAwait.ClusterName,
+				WithSpecSpaceRole("admin"),
+				WithSpecMasterUserRecord("invalidMUR"), // we set an invalid MUR
+				WithNamespace(GetDefaultNamespace(space.Status.ProvisionedNamespaces)),
+			)
+
+			// then
+			require.NoError(t, err)
+			// wait for spacebinding request status to be set
+			_, err = memberAwait.WaitForSpaceBindingRequest(t, types.NamespacedName{Namespace: spaceBindingRequest.GetNamespace(), Name: spaceBindingRequest.GetName()},
+				UntilSpaceBindingRequestHasConditions(spacebindingrequesttestcommon.UnableToCreateSpaceBinding("unable to get MUR: MasterUserRecord.toolchain.dev.openshift.com \"invalidMUR\" not found")),
+			)
+			require.NoError(t, err)
+		})
+	})
+}
+
+func TestUpdateSpaceBindingRequest(t *testing.T) {
+	// given
+	t.Parallel()
+	// make sure everything is ready before running the actual tests
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+
+	t.Run("update space binding request SpaceRole", func(t *testing.T) {
+		// when
+		space, spaceBindingRequest, _ := NewSpaceBindingRequest(t, awaitilities, memberAwait, hostAwait, "contributor")
+		_, err := memberAwait.UpdateSpaceBindingRequest(t, types.NamespacedName{Namespace: spaceBindingRequest.Namespace, Name: spaceBindingRequest.Name},
+			func(s *toolchainv1alpha1.SpaceBindingRequest) {
+				s.Spec.SpaceRole = "admin" // set to admin from contributor
+			},
+		)
+		require.NoError(t, err)
+
+		//then
+		// wait for both SpaceBindingRequest and SpaceBinding to have same SpaceRole
+		spaceBindingRequest, err = memberAwait.WaitForSpaceBindingRequest(t, types.NamespacedName{Namespace: spaceBindingRequest.GetNamespace(), Name: spaceBindingRequest.GetName()},
+			UntilSpaceBindingRequestHasConditions(spacebindingrequesttestcommon.Ready()),
+			UntilSpaceBindingRequestHasSpecSpaceRole("admin"), // has admin role
+			UntilSpaceBindingRequestHasSpecMUR(spaceBindingRequest.Spec.MasterUserRecord),
+		)
+		require.NoError(t, err)
+		_, err = hostAwait.WaitForSpaceBinding(t, spaceBindingRequest.Spec.MasterUserRecord, space.Name,
+			UntilSpaceBindingHasMurName(spaceBindingRequest.Spec.MasterUserRecord),
+			UntilSpaceBindingHasSpaceName(space.Name),
+			UntilSpaceBindingHasSpaceRole("admin"), // has admin role
+			UntilSpaceBindingHasLabel(toolchainv1alpha1.SpaceBindingRequestLabelKey, spaceBindingRequest.GetName()),
+			UntilSpaceBindingHasLabel(toolchainv1alpha1.SpaceBindingRequestNamespaceLabelKey, spaceBindingRequest.GetNamespace()),
+		)
+		require.NoError(t, err)
+	})
+
+	t.Run("update space binding request MasterUserRecord", func(t *testing.T) {
```

Comment:
I'm wondering how much we want to support this scenario. I would love to make the MUR immutable, which would require an admission webhook of course :-/. 
The admission webhook could also do all the other validations we were talking about - it would make much more sense to have it in a webhook than in a controller.

Reply:
Makes sense, I haven't considered that the name doesn't change! 
I'm now updating the mur name in the spacebinding spec and label but not changing the name, so maybe we should not allow this sort of update. But right now the user/client has the ability to update this field, so wanted to make sure the code works as "expected"

Reply:
sure, that was more as a follow up task